lib/services/watcher.go

```diff
@@ -883,36 +910,37 @@ func (p *databaseCollector) defineCollectorAsInitialized() {
 	})
 }
 
-// processEventAndUpdateCurrent is called when a watcher event is received.
-func (p *databaseCollector) processEventAndUpdateCurrent(ctx context.Context, event types.Event) {
-	if event.Resource == nil || event.Resource.GetKind() != types.KindDatabase {
-		p.Log.Warnf("Unexpected event: %v.", event)
-		return
-	}
+// processEventsAndUpdateCurrent is called when a watcher event is received.
+func (p *databaseCollector) processEventsAndUpdateCurrent(ctx context.Context, events []types.Event) {
 	p.lock.Lock()
 	defer p.lock.Unlock()
-	switch event.Type {
-	case types.OpDelete:
-		delete(p.current, event.Resource.GetName())
-		select {
-		case <-ctx.Done():
-		case p.DatabasesC <- resourcesToSlice(p.current):
-		}
-	case types.OpPut:
-		database, ok := event.Resource.(types.Database)
-		if !ok {
-			p.Log.Warnf("Unexpected resource type %T.", event.Resource)
-			return
-		}
-		p.current[database.GetName()] = database
-		select {
-		case <-ctx.Done():
-		case p.DatabasesC <- resourcesToSlice(p.current):
+	for _, event := range events {
+		if event.Resource == nil || event.Resource.GetKind() != types.KindDatabase {
+			p.Log.Warnf("Unexpected event: %v.", event)
+			continue
 		}
+		switch event.Type {
+		case types.OpDelete:
+			delete(p.current, event.Resource.GetName())
+			select {
+			case <-ctx.Done():
+			case p.DatabasesC <- resourcesToSlice(p.current):
```

Comment:
Likewise, do we need to send an update to `DatabasesC` more than once per batch?